lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/restart/DFSEvaluatorPreserver.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runtime.yarn.driver.restart;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.reef.driver.parameters.FailDriverOnEvaluatorLogErrors;
+import org.apache.reef.exception.DriverFatalRuntimeException;
+import org.apache.reef.runtime.common.driver.EvaluatorPreserver;
+import org.apache.reef.runtime.common.driver.evaluator.EvaluatorManager;
+import org.apache.reef.tang.annotations.Parameter;
+
+import javax.inject.Inject;
+import java.io.*;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * An Evaluator Preserver that uses the DFS on YARN.
+ */
+public final class DFSEvaluatorPreserver implements EvaluatorPreserver {
+  private static final Logger LOG = Logger.getLogger(DFSEvaluatorPreserver.class.getName());
+
+  private static final String ADD_FLAG = "+";
+
+  private static final String REMOVE_FLAG = "-";
+
+  private final boolean failDriverOnEvaluatorLogErrors;
+
+  private DFSEvaluatorLogWriter writer;
+
+  private Path changeLogLocation;
+
+  private FileSystem fileSystem;
+
+
+  @Inject
+  private DFSEvaluatorPreserver(@Parameter(FailDriverOnEvaluatorLogErrors.class)
+                                final boolean failDriverOnEvaluatorLogErrors) {
+
+    this.failDriverOnEvaluatorLogErrors = failDriverOnEvaluatorLogErrors;
+
+    try {
+      final org.apache.hadoop.conf.Configuration config = new org.apache.hadoop.conf.Configuration();
+      this.fileSystem = FileSystem.get(config);
+      this.changeLogLocation =
+          new Path("/ReefApplications/" + EvaluatorManager.getJobIdentifier() + "/evaluatorsChangesLog");
```

Comment:
We have `org.apache.reef.client.DriverConfiguration.DRIVER_JOB_SUBMISSION_DIRECTORY` which gets bound to `org.apache.reef.driver.parameters.DriverJobSubmissionDirectory`. I believe that is what is being used in the future. Hence, it would be safe to use that named parameter here as well, e.g. `DriverJobSubmissionDirectory/reef/evaluatorsChangeLog`.